### DISCUSSION TITLE
Push/Sync doesn't show updates in CODE

### DISCUSSION THREAD

**Discussion by RogerMathews** [community/community#22389] [New to GitHub]
<p>I am using GitHub with Visual Studio on my personal Laptop and work Laptop.  I created a simple change on one laptop, committed locally, then SYNC.  On the next laptop did a sync and the code from the first laptop showed up but when I go into the GitHub portal under the code tab the small test code change isn’t there and the date is not updated.</p>
<p>I am struggling to understand the basic workflow of how to use GitHub.  I am using a single GitHub account between both laptops.</p>
<p>I do not have any additional branches, just the root “main”.</p>
<p>Any ideas?</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi Roger,</p>
<p>The workflow is simple:</p>
<ul>
<li>do some change to your code and save</li>
<li>add into the stage area =&gt; on your integrated vscode terminal you can type <code>git add --all</code> or if you want to use a gui on vscode the third tab do the job:<br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/1/9143d058ee97fdfa5326e3ba071cc889c68421fc.png" data-download-href="/uploads/short-url/kJ4wteop5dmz458nVBs1z1X9Akc.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/12409541/181070708-2d872b19-ba55-477b-a7df-0137d248d635.png" alt="image" data-base62-sha1="kJ4wteop5dmz458nVBs1z1X9Akc" width="265" height="250" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/1/9143d058ee97fdfa5326e3ba071cc889c68421fc_2_265x250.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/1/9143d058ee97fdfa5326e3ba071cc889c68421fc_2_397x375.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/9/1/9143d058ee97fdfa5326e3ba071cc889c68421fc_2_530x500.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/1/9143d058ee97fdfa5326e3ba071cc889c68421fc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">590×556 29 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<br>
if you click on the + symbol you tell git the file you want to save on the history of the project.</li>
<li>then commit this changes to confirm the change you have stage =&gt; if you are using a terminal you can run <code>git commit -m 'your message'</code> or if you are using the gui on vscode the message is the first text block of the tab:<br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/9/c9a0638334abfa6a348d356e44daffdfad5e7193.png" data-download-href="/uploads/short-url/sLFBncrjEU6fASdrF5aOJp0nMTp.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/12409541/181070709-f07fa77d-fb7e-4fd6-ba1b-873498c91034.png" alt="image" data-base62-sha1="sLFBncrjEU6fASdrF5aOJp0nMTp" width="284" height="250" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/9/c9a0638334abfa6a348d356e44daffdfad5e7193_2_284x250.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/9/c9a0638334abfa6a348d356e44daffdfad5e7193_2_426x375.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/9/c9a0638334abfa6a348d356e44daffdfad5e7193_2_568x500.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/9/c9a0638334abfa6a348d356e44daffdfad5e7193_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">588×516 32.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<br>
Remember that a message in the commit is mandatory. Once you have write the message just click on the check symbols above.</li>
<li>at last push the change to github =&gt; even for this you could use the terminal command <code>git push origin main</code> or the gui in vscode. In this case the button you need to click is at the bottom of the page:<br>
<img src="https://user-images.githubusercontent.com/12409541/181070710-74f9f252-6606-4b85-ac23-b5c1ec87bed1.png" alt="image" data-base62-sha1="rWFJzXmD8PFYM1YEbJwDwUbRWb1" width="276" height="116"><br>
Click the button next to the branch name <code>main</code> and you could see the code uploaded to github if there’s no error.</li>
<li>in the other pc you need to update the repository and you can do it by simply run the pull command <code>git pull origin main</code> or if you use the vscode gui just click the same button you click for the push.</li>
</ul>
<p>This is it, hope this help :blush:</p>
<p>If I can advise I would like to propose you to do the official course that you could find in <a href="https://lab.github.com/" rel="noopener nofollow ugc">https://lab.github.com/</a> or if you are a book reader like me you can find this one very interesting <a href="https://git-scm.com/book/en/v2" class="inline-onebox" rel="noopener nofollow ugc">Git - Book</a> to start using git like a jedi.</p>
<p>Let me if this solves your problem :relaxed:</p>

**Comment by RogerMathews** [comment]
<p>This was helpful, thank you.  I also found an issue in my 10-year old solution that was giving me grief for one of my projects as it relates to GitHub.</p>
<p>All is good now.</p>
